### PR TITLE
feat: added GtfsNetworkSchema

### LINKED ISSUE #1640: Add remaining GTFS Features to the Validator
### Describe the problem

The spec team at MobilityData has started to structure the spec into features (e.g Wheelchair Accessibility, Pathways) to 1) make it easier for newcomers to the spec to understand what use cases can be addressed with it and 2) make it easy for governments and operators to require specific features in contracts and regulatory requirements. 

[In progress spreadsheet here](https://docs.google.com/spreadsheets/d/1kpbKOzlHtsJIPo3B4ABYu-Nxwvqutjcdcvgpa0wZvtA/edit#gid=1997312156)

We started adding these to the validator metadata report so the community can see what features are in different feeds and to also help with our own internal analytics. The first iteration of this list tracks **the presence of features** in the feed, as in if there is 1 occurrence, we count it. 

Now that the list of features and its logic are almost finalized, we need to 1) add new features that are missing in the validator and 2) do some minor change of logic. 

### Proposed solution

[FeedMetadata](https://github.com/MobilityData/gtfs-validator/blob/master/main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java)

Logic should check values so there has to be 1 record (not just the column headers) for each feature

- [x] ~~Change Bikes Allowed = if at least 1 value of bikes_allowed in trips.txt defined. Remove check for values 2 or 3.~~ (already done) 
- [x] Change Zone-Based Fares = if 1 record in areas.txt
      - [MBTA: Example feed](https://cdn.mbta.com/MBTA_GTFS.zip)
- [x] Change name of Frequency-Based Trips feature to Frequencies
    - [Bullrunner: Example feed download link](https://github.com/CUTR-at-USF/bullrunner-gtfs-realtime-generator/blob/master/bullrunner-gtfs.zip)
- [x] Add Route-Based Fares = If at least 1 value of network_id in routes.txt defined OR 1 record in networks.txt
      - [MBTA: Example feed](https://cdn.mbta.com/MBTA_GTFS.zip)
- [x] Add Time-Based Fares = If 1 record in timeframes.txt
- [x] Add Transfer Fares = If 1 record in fare_transfer_rules.txt
      - [MBTA: Example feed](https://cdn.mbta.com/MBTA_GTFS.zip)
- [x] Add Pathways (extra) = If 1 occurrence of max_slope OR min_width OR length OR stair_count in pathways.txt
      - [MBTA: Example feed](https://cdn.mbta.com/MBTA_GTFS.zip)
- [x] Add Levels = If 1 record in levels.txt
      - [MBTA: Example feed](https://cdn.mbta.com/MBTA_GTFS.zip)

### Alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by qcdyx**
**Summary:**

Part 3 of #1640 
Closes #1640 

**Expected behavior:** 
added a new GtfsNetworkSchema in code which represents the networks.txt.
https://docs.google.com/spreadsheets/d/1iyb8SCRxEXiK8-PHq0_Ut1_QUbeUEbQF6yZiA54WztU/edit#gid=1906649886

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Add or update any needed [documentation](https://github.com/MobilityData/gtfs-validator/tree/master/docs) to the repo 
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsNetworkSchema.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.mobilitydata.gtfsvalidator.table;
+
+import org.mobilitydata.gtfsvalidator.annotation.*;
+
+@GtfsTable("networks.txt")
+@Required
+public interface GtfsNetworkSchema extends GtfsEntity {
+  @FieldType(FieldTypeEnum.ID)
+  @PrimaryKey
+  @Required
```
Comment (emmambd) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsNetworkSchema.java:10]: Should be optional field, not required, and add conditionally forbidden logic in a separate PR. Forbidden if network_id exists in [routes.txt](https://gtfs.org/schedule/reference/#routestxt). cc @jcpitre

Comment (qcdyx) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsNetworkSchema.java:9]: As discussed in the standup, @jcpitre can take care of this so I can finish the load test for Mobility Database API!

**Comment by emmambd** [comment]
Screenshot 2024-01-25 at 11 04 58 AM

Route-based fares appears! LGTM on a QA side.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsNetworkSchema.java (hunk 1 above)
Comment (jcpitre) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsNetworkSchema.java:9]: Delegated to another PR

**Review by jcpitre — APPROVED** [review]

File: main/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsNetworkSchema.java (hunk 1 above)
Comment (davidgamez) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsNetworkSchema.java:9]: Before merging let's remove the `Required` annotations.

Comment (qcdyx) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsNetworkSchema.java:9]: It's already removed. Ready to be merged.

**Pull request merged**